src/core/connection.c

```diff
@@ -6399,3 +6423,34 @@ QuicConnDrainOperations(
 
     return HasMoreWorkToDo;
 }
+
+#ifdef QUIC_WIRESHARK_SUPPORT
+void
+QuicConnSslKeyLogFileWrite(
+    _In_ QUIC_CONNECTION* Connection,
+    _In_z_ const char* Line
+    )
+{
+    if (Connection->SslKeyLogBuffer != NULL && Connection->SslKeyLogBytesRemaining > 0) {
+        size_t LineLen = strlen(Line);
+        //
+        // Append newline on previous log line.
+        //
+        if (Connection->SslKeyLogBytesRemaining > 0) {
+            *Connection->SslKeyLogBuffer = '\n';
+            Connection->SslKeyLogBuffer++;
+            Connection->SslKeyLogBytesRemaining--;
+        }
+        memcpy(Connection->SslKeyLogBuffer, Line, min(LineLen, Connection->SslKeyLogBytesRemaining));
```

Comment:
change `min(LineLen, Connection->SslKeyLogBytesRemaining)` to  `min(LineLen, Connection->SslKeyLogBytesRemaining - 1)` to account for NULL at the end.